src/components/spatial/SpatialSubscriber.js

```diff
@@ -116,27 +126,41 @@ export default function SpatialSubscriber(props) {
   const [raster, imageLayerLoaders] = useRasterData(
     loaders, dataset, setItemIsReady, addUrl, false,
     autoImageLayers => setAutoLayers(prev => (
-      { [dataset]: [...(prev[dataset] || []), ...autoImageLayers] }
+      // This prevents old updates from overriding the current dataset.
+      // The previous state must be for this dataset, otherwise it's an old update.
+      Object.keys(prev).includes(dataset)
+        ? { [dataset]: [...(prev[dataset] || []), ...autoImageLayers] }
+        : prev
     )),
   );
-
   // Try to set up the layers array automatically if null or undefined.
   useEffect(() => {
-    if (isReady && !layers && autoLayers[dataset]) {
-      setLayers(sortLayers(autoLayers[dataset]));
-    } else if (isReady && layers) {
-      // Layers were defined, but check whether channels for each layer were also defined.
-      // If channel / slider / domain definitions are missing, initialize in automatically.
-      initializeLayerChannelsIfMissing(layers, imageLayerLoaders)
-        .then(([newLayers, didInitialize]) => {
-          if (didInitialize) {
-            // Channels were only partially defined.
-            setLayers(newLayers);
-          }
-        });
+    // Check if the autoLayers have a layer for each spatial layer loader type.
+    const areAutoLayersComplete = Object.keys(loaders[dataset].loaders)?.every(
+      loaderType => !SPATIAL_LAYER_TYPES.includes(loaderType)
+        || (
+          autoLayers[dataset] && autoLayers[dataset].filter(
+            layer => layer.type === loaderType,
+          ).length > 0
+        )
+    );
```

Comment:
Due to the change of initializing `autoLayers` with something instead of nothing, we had a situation where the `autoLayers` could be "incomplete" so this makes sure they are now complete before trying anything.